core/src/test/scala/org/apache/spark/util/collection/SorterSuite.scala

```diff
@@ -74,6 +74,103 @@ class SorterSuite extends SparkFunSuite with Logging {
     (0 to data.length - 2).foreach(i => assert(data(i) <= data(i + 1)))
   }
 
+  test("java.lang.ArrayIndexOutOfBoundsException in TimSort") {
```

Comment:
Could we revert this commit or improve the test time?

![Screen Shot 2019-04-10 at 2 09 13 PM](https://user-images.githubusercontent.com/11567269/55913793-50312500-5b9a-11e9-907b-b5acd8492697.png)

Reply:
How about to speed up the sort itself?

Reply:
The time can be improved if we remove result checking. Actually the test checks that TimSort does not fail with the exception `ArrayIndexOutOfBoundsException`

Reply:
I'm OK with that. Checking correctness is nice but it's costly given how huge the array is. I think we already tried to optimize it. Other tests are evaluating correctness of the sort, so kinda OK to not do that here too. That said I don't know how much removing that check helps? worth trying

Reply:
Here is the PR https://github.com/apache/spark/pull/24343 which speeds up the test ~ 10 times at least on my laptop.